### PR TITLE
feat: ConfigService cache record stats

### LINKED ISSUE #5246: ConfigServiceWithCache 新增recordStats功能
**你的特性请求和某个问题有关吗？请描述**

目前Apollo-ConfigService为了提升性能，会选择开启配置缓存功能ConfigServiceWithCache，但是目前对于缓存的size、命中率、淘汰数量等数据没有很好的监控，无法及时处理一下问题

**清晰简洁地描述一下你希望的解决方案**

希望ConfigServiceWithCache能新增一个监控缓存的功能，此功能可通过增加一个开关的方式手动开启，默认关闭（兼容性）

**清晰简洁地描述一下这个特性的备选方案**

**其它背景**

BTW：如果觉得这个特性合理，我可以实现这个方案

### PR DESCRIPTION
**Pull request by youngzil**
## What's the purpose of this PR

ConfigService cache record stats

When the service is just started, we visit `curl 127.0.0.1:8080/prometheus | grep -E 'config_cache|config_id_cache'`

image

After some pull configuration requests, we visit `curl 127.0.0.1:8080/prometheus | grep -E 'config_cache|config_id_cache'`, and the cache monitoring indicators start recording data

image

## Which issue(s) this PR fixes:
Fixes #5246

## Brief changelog

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/apolloconfig/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [x] Update the [`CHANGES` log](https://github.com/apolloconfig/apollo/blob/master/CHANGES.md).

## Release Notes

- **New Features**
  - Introduced a method to enable cache statistics in configuration settings.
  - Enhanced cache initialization with monitoring capabilities for better performance insights.
  - Added a global search feature for administrators to easily find values.
  - Comprehensive instructions for deploying Apollo using Docker and Kubernetes.

- **Bug Fixes**
  - Resolved issues with duplicate comments and blank lines in configuration management.
  - Fixed a link issue where some published items were missing.

- **Documentation**
  - Updated the distributed deployment guide with detailed instructions for various deployment scenarios.
  - Expanded sections on environment preparation, database requirements, and network configuration for clarity and usability.
  - Added new sections describing the cache statistics function and its configuration.

These updates improve user experience and provide more control over configuration and deployment processes.

### CODE REVIEW

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/config/ConfigServiceWithCache.java [hunk 1]
```diff
@@ -64,6 +67,9 @@ public class ConfigServiceWithCache extends AbstractConfigService {
   private final ReleaseMessageService releaseMessageService;
   private final BizConfig bizConfig;
 
+  @Autowired
+  private MeterRegistry meterRegistry;
```
Comment (nobodyiam) [inline apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/config/ConfigServiceWithCache.java:71]: Prefer constructor injection over autowired for better unit test compatibility.

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/config/ConfigServiceWithCache.java [hunk 2]
```diff
@@ -64,6 +67,9 @@
...
+
```
[the remark this answers is not in the document]
Comment (nobodyiam) [reply apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/config/ConfigServiceWithCache.java:72]: Wow, this is a much better explanation than mine.

File: docs/en/deployment/distributed-deployment-guide.md [hunk 3]
```diff
@@ -1521,6 +1521,17 @@ This configuration takes effect when config-service.cache.enabled is set to true
...
+#### 3.2.3.2 config-service.cache.stats.enabled - Whether to enable caching metric statistics function
+> For versions 2.4.0 and above
+
+> `config-service.cache.stats.enabled` The adjustment configuration must be restarted config service to take effect.
+
+This configuration works when `config-service.cache.stats.enabled` is true, it is used to control the opening of the cache statistics function.  
+The default is false, that is, it will not enable the cache statistics function, when it is set to true, it will enable the cache metric statistics function, see the reference index[Monitoring related-5.2 Metrics](en/design/apollo-design#5.2-Metrics)
+
+
+
```
Comment (nobodyiam) [inline docs/en/deployment/distributed-deployment-guide.md:1534]: Too many empty lines?

**Review by nobodyiam — COMMENTED** [review]
How can the user view the metrics after enabling the feature?

**Comment by youngzil** [comment]
@nobodyiam I have solved these problems, please review again

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/config/ConfigServiceWithCache.java [hunk 4]
```diff
@@ -44,6 +46,7 @@
 import java.util.concurrent.TimeUnit;
 
 import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (nobodyiam) [inline apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/config/ConfigServiceWithCache.java:49]: Is it needed?

**Review by nobodyiam — COMMENTED** [review]
Would you please also post a screenshot in the pr description to show the actual effect?

**Review by nobodyiam — APPROVED** [review]
LGTM

**Pull request merged**